reana_db/models.py

```diff
@@ -387,6 +398,18 @@ def get_full_workflow_name(self):
         """Return full workflow name including run number."""
         return "{}.{}".format(self.name, str(self.run_number))
 
+    def get_workspace_disk_usage(self, summarize=False, block_size=None):
+        """Retrieve disk usage information of a workspace."""
+        disk_usage_info = get_disk_usage(self.workspace_path, summarize, block_size)
+        # create pairs of (size, filename)
+        filesize_pairs = list(zip(disk_usage_info[::2], disk_usage_info[1::2]))
+        filesizes = []
+        for filesize_pair in filesize_pairs:
+            size, name = filesize_pair
+            # trim workspace path in every file name
+            filesizes.append({"name": name[len(self.name) :], "size": size})
```

Comment:
Since this logic is specific to `du`'s output, maybe we can do it inside `get_disk_usage`? Wha do you think of something like this:
- Always returning a list of dicts with `name` and `value`
  - In case the of `summarize==True` it can be a single item list
  - In case of `summarize==False` it can be the list we build here

Reply:
(in that case, we would need to improve [`get_disk_usage`](https://github.com/reanahub/reana-commons/pull/208/files#diff-c652280f4f14efe4c398627ef0fb5da5R193) docstring so it is clearly documented)

Reply:
I've separated this peace of functionality from the `get_disk_usage` since it's not needed for the `get_user_disk_usage` (User model). But you are right, having this code inside the Workflow model does not make too much sense. I will amend it with your suggestions